### DISCUSSION TITLE
Boto3 (Lib for s3 media backend) is not included in Dockerfiles

### DISCUSSION THREAD

**Discussion by MaxRink** [netbox-community/netbox-docker#416] [Ideas]
As it is right now boto3 isnt included in the dockerfile, which makes using S3 as filebackend impossible without building your own image.
Given the fact that S3 would be very usefull in the container ( as that makes scaling to multiple instances easier), is there any interest in including it in the image?

**Comment by cimnine** [comment]
I think that could actually e a good idea.

**Comment by tobiasge** [comment]
We should include this. Django-storages is not usable without a backend. In the `setup.cfg` of `django-storages` I found these requirements:
```
azure =
    azure-storage-blob >=1.3.1,<12.0.0
boto3 =
    boto3 >= 1.4.4
dropbox =
    dropbox >= 7.2.1
google =
    google-cloud-storage >= 1.15.0
libcloud =
    apache-libcloud
sftp =
    paramiko
```

I think we should include the backends for the most common service. My suggestion would be: `boto3`, `azure-storage-blob` and `google-cloud-storage`.

**Reply by cimnine** [reply]
Why not just all of them?

**Reply by tobiasge** [reply]
I just thought we could keep the images smaller, but we can also add all. I think the best way would be to use the extra requirements. That we don't have to keep track of the versions:
`django-storages[azure,boto3,dropbox,google,libcloud,sftp]==1.11.1` in our `requirements-container.txt`

**Comment by tobiasge** [comment]
Pull request for this: #421
